nbqa/__main__.py

```diff
@@ -91,9 +91,42 @@ def _get_notebooks(root_dir: str) -> Iterator[Path]:
     )
 
 
-def _get_all_notebooks(root_dirs: List[str]) -> Iterator[Path]:
+def _filter_by_include_exclude(
+    notebooks: Iterator[Path],
+    include: Optional[str],
+    exclude: Optional[str],
+) -> Iterator[Path]:
     """
-    Get generator with all notebooks passed in via the command-line.
+    Include files which match include, exclude those matching exclude.
+
+    notebooks
+        Notebooks (not directories) to run code quality tool on.
+    include:
+        Global file include pattern.
+    exclude:
+        Global file exclude pattern.
+
+    Returns
+    -------
+    Iterator
+        Notebooks matching include and not matching exclude.
+    """
+    include = include or ""
+    exclude = exclude or "^$"
+    include_re, exclude_re = re.compile(include), re.compile(exclude)
```

Comment:
Was about to write pretty much the same comment yday, but when I checked our supported tolls I saw that only `flake8`, `mypy` and `isort` use glob patterns (`isort` even has an extra flag for it). While `pre-commit`, `pylint` and `black` use RegEx.
So we already have a mixed way to exclude files in the QA ecosystem, and since `pre-commit` is the recommendet way to use `nbqa`, people will have to get more familiar with RegEx anyway.
Thus I did write a comment sugesting glob patterns instead. 😜

Reply:
Just saw @MarcoGorelli wrote pretty much the same train of thought below 🤣

Reply:
> Was about to write pretty much the same comment yday, but when I checked our supported tolls I saw that only flake8, mypy and isort use glob patterns (isort even has an extra flag for it). While pre-commit, pylint and black use RegEx.

I just looked at flake8 since it was mentioned in the issue #429 and hence raised this question. But going with pre-commit way is the right thing to do given how well nbqa is made for pre-commit usage.